### PR TITLE
added documentation on meshsync,operator and database

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

**Description**

Documentation updated for:
- Meshery Operator
- MeshSync
- Meshery Database
- Meshery Broker

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: docs/pages/concepts/architecture/architecture.md [hunk 1]
```diff
@@ -54,6 +55,15 @@ Meshery Operator is the multi-service mesh operator (a Kubernetes custom control
 [![Meshery Operator and MeshSync]({{ site.baseurl }}/assets/img/architecture/meshery-operator-and-meshsync.svg
 )]({{ site.baseurl }}/assets/img/architecture/meshery-operator-and-meshsync.svg)
 
+See the [**Operator**]({{ site.baseurl }}/architecture/operator) section for more information on the function of an operator and [**MeshSync**]({{ site.baseurl }}/architecture/meshsync) section for more information on the function of meshsync.
+
+## Meshery Database
+Meshery Database is responsible to persist and manage the state of meshery and its components.
```
Comment (leecalcote) [inline docs/pages/concepts/architecture/architecture.md:61]:
```suggestion
Meshery Server's database is responsible for collecting and centralizing the state of all elements under management, including infrastructure, application, and Meshery's own components. Meshery's database, while persisted to file, is treated as a cache.

```

File: docs/pages/concepts/architecture/architecture.md [hunk 2]
```diff
@@ -54,6 +55,15 @@ Meshery Operator is the multi-service mesh operator (a Kubernetes custom control
...
+See the [**Operator**]({{ site.baseurl }}/architecture/operator) section for more information on the function of an operator and [**MeshSync**]({{ site.baseurl }}/architecture/meshsync) section for more information on the function of meshsync.
+
+## Meshery Database
+Meshery Database is responsible to persist and manage the state of meshery and its components.
+
+[![Meshery Database]({{ site.baseurl }}/assets/img/architecture/meshery-database.svg)]({{ site.baseurl }}/assets/img/architecture/meshery-database.svg)
+
+See the [**Database**]({{ site.baseurl }}/architecture/database) section for more information on the function of the database.
+
 ### **Network Ports**
```
Comment (leecalcote) [inline docs/pages/concepts/architecture/architecture.md:67]: @kumarabd please review existing and capture new network ports for database (internal only), operator, broker.

File: docs/pages/concepts/architecture/broker.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+title: Broker
+permalink: concepts/architecture/broker
+type: concepts
+redirect_from: architecture/broker
+abstract: "Meshery broker component fascilitates data streaming between kubernetes cluster components and outside world."
+language: en
+list: include
+---
+
+Broker is a custom kubernetes controller that empowers data streaming across different components of meshery, which are either inside or outside of the kubernetes cluster.
```
Comment (leecalcote) [inline docs/pages/concepts/architecture/broker.md:12]:
```suggestion
Broker is a custom Kubernetes controller that provides data streaming across independent components of Meshery whether those components are running inside or outside of the Kubernetes cluster.
```

File: docs/pages/concepts/architecture/broker.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+### Broker FAQs
```
Comment (leecalcote) [inline docs/pages/concepts/architecture/broker.md:14]: - How many Brokers can run? 
- What does an HA configuration look like? 
- What stateful characteristics does the Broker have?
- How do I know if the Broker is working? How do I troubleshoot the Broker?

File: docs/pages/concepts/architecture/database.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+title: Database
+permalink: concepts/architecture/database
+type: concepts
+redirect_from: architecture/database
+abstract: "Meshery offers support for internal caching with the help of file databases. This has been implemented with several libraries that supports different kinds of data formats."
+language: en
+list: include
+---
+
+## What are Meshery Adapters?
```
Comment (leecalcote) [inline docs/pages/concepts/architecture/database.md:12]:
```suggestion
## What is the Meshery Database?
```

File: docs/pages/concepts/architecture/database.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+## Components Involved:
+
+Meshery Database has several kinds of database implementations to support various usecases. They are listed below:
+{% assign sorted = site.adapters | sort: "project_status" | reverse %}
+
+| Component      | Library                               |
+| :------------- | :------------------------------------ |
+| Bitcask        | github.com/prologic/bitcask           |
+| SQLite         | gorm.io/gorm, gorm.io/driver/sqlite   |
```
Comment (leecalcote) [inline docs/pages/concepts/architecture/database.md:24]: The database diagram can be added here.

File: docs/pages/concepts/architecture/meshsync.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+title: MeshSync
+permalink: concepts/architecture/meshsync
+type: concepts
+redirect_from: architecture/meshsync
+abstract: "Meshery offers support for kubernetes cluster/service mesh state synchronization with the help of MeshSync."
+language: en
+list: include
+---
+
+MeshSync is a custom kubernetes controller that empowers synchronization of the state of the cluster with Meshery server.
```
Comment (leecalcote) [inline docs/pages/concepts/architecture/meshsync.md:12]:
```suggestion
MeshSync is a custom Kubernetes controller that provides tiered discovery and continual synchronization with Meshery Server as to the state of the Kubernetes cluster, service meshes, and their workloads.
```

File: docs/pages/concepts/architecture/architecture.md (hunk 2 above)
Comment (kumarabd) [reply docs/pages/concepts/architecture/architecture.md:68]: @leecalcote It is already been captured for broker. And we don't need any for the database.

File: docs/pages/concepts/architecture/database.md (hunk 6 above)
Comment (kumarabd) [reply docs/pages/concepts/architecture/database.md:26]: done

File: docs/pages/concepts/architecture/broker.md (hunk 4 above)
Comment (kumarabd) [reply docs/pages/concepts/architecture/broker.md]: Done done..

File: docs/pages/concepts/architecture/architecture.md (hunk 2 above)
Comment (leecalcote) [reply docs/pages/concepts/architecture/architecture.md:68]: Ok. Rather, the operator reuses the existing https port existing between K8s and Meshery.

**Review by leecalcote — APPROVED** [review]

**Pull request merged**